### PR TITLE
[CI][VSTS] Do not run old macOS on PRs.

### PR DESCRIPTION
**Pull request by mandel-macaque**
fixes: https://github.com/xamarin/maccore/issues/2395

### CODE REVIEW

File: tools/devops/automation/templates/mac/stage.yml [hunk 1]
```diff
@@ -19,8 +19,7 @@ stages:
   dependsOn:
   - build_packages
   # we need to have the pkgs built and the device sets to be ran, that is decided via the labels or type of build during the build_packages stage
-  # condition: and(succeeded(), eq (stageDependencies.build_packages.build.outputs['configuration.RunMacTests'], 'True')) 
-  condition: succeeded()
+  condition: and(succeeded(), eq(dependencies.build_packages.outputs['build.configuration.RunDeviceTests'], 'True'))
```
Comment (mandel-macaque) [inline tools/devops/automation/templates/mac/stage.yml:22]: Same line as in the device tests.

**Review by spouliot — APPROVED** [review]

**Review by rolfbjarne — COMMENTED** [review]
Can't we make PRs build packages instead so that we get the old mac tests as well on PRs?

**Comment by mandel-macaque** [comment]
@rolfbjarne yes, I've added https://github.com/xamarin/maccore/issues/2396 and will follow after this.

**Pull request merged**